wordpress-mobile/WordPress-iOS
# Improves the coloring of the search fields throughout the App.

This change was spawned by a chat with @osullivanchris.

This PR improves the coloring of the search fields in dark mode throughout the App.  It changes the background to match the navigation bar color, and it changes the search-query-tex-color to be the default (black for light mode, and white for dark mode).

## Search Field Placeholder Text Coloring

| Before | After |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/1836005/116067207-3a133600-a689-11eb-9cc4-2a96c3a52d1f.png" width=300> | <img src="https://user-images.githubusercontent.com/1836005/116110317-dfdc9a00-a6b5-11eb-9295-091aaae67ef6.png" width=300> |
| <img src="https://user-images.githubusercontent.com/1836005/116068595-b2c6c200-a68a-11eb-9b96-3abb7fdbb1ea.png" width=300> | <img src="https://user-images.githubusercontent.com/1836005/116110409-f4209700-a6b5-11eb-84b2-2a6b1fe8977d.png" width=300> |


<img src="https://user-images.githubusercontent.com/1836005/116067287-4e573300-a689-11eb-85f7-f0e8d6637b7d.png" width=300> |

## Search Field Input Text Coloring

| Before | After |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/1836005/116068043-0e448000-a68a-11eb-8bc5-204948d4abc7.png" width=300> | <img src="https://user-images.githubusercontent.com/1836005/116067557-88c0d000-a689-11eb-903f-f470561af9fc.png" width=300> |
| <img src="https://user-images.githubusercontent.com/1836005/116068544-a3e00f80-a68a-11eb-8df4-63c12940be9b.png" width=300> | <img src="https://user-images.githubusercontent.com/1836005/116067568-8ceced80-a689-11eb-82be-0dccd172ac7f.png" width=300> |

## To test:

Test in light and dark mode.

1. Go to any place where you expect a search field.
2. Make sure the color of the search field matches the color of the navigation bar.

## Regression Notes

1. Potential unintended areas of impact

Any search field that doesn't have a navigation bar.  Below is a list of places where the search bar is configured using the changes in this PR.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I tested each of those fields in-app.

✅ AztecPostViewController.swift: this isn't easy to access because it's a search bar for the media picker in the classic editor.  You need to install the classic editor plugin in your site, create a post, and then edit it using WPiOS and add an image from the  WordPress media library.
✅ BlogListViewController.m: the behavior here seems to be what I'd expect, but I do think it's worth mentioning when you give focus to the search field, the background changes color in a slightly unexpected way (code wise).
✅ BlogSelectorViewController.m: this is the site picker.  There's a divider between the VC title and the search bar that I wasn't able to remove without risking introducing regressions.  This was discussed with both @frosty and @osullivanchris and we agreed it could be left like this for now.
✅ SiteTagsViewController.swift: My Site > Site Settings > Tags.  You need to have at least 1 tag created to see the search field.  There's some divider weirdness worth mentioning here, although it wasn't introduced by my changes.
✅ LanguageSelectorViewController.swift: you need a wp.com site to find this one.  My Site > Site Settings > Language.  Same background changing effect as reported for the blog list.
✅ SiteIconPickerPresenter.swift: My Site > Tap Site Icon > Change Site Icon > "WordPress Media" album
✅ TenorPicker.swift: New post > Add media > Add image > Free GIF Library
✅ StockPhotosPicker.swift: New post > Add media > Add image > Free Photo Library
✅ MediaLibraryViewController.swift: My Site > Media
✅ ParentPageSettingsViewController.swift: My Site > Site Pages > Tap on "..." for an existing page > Set Parent
✅ PluginDirectoryViewController.swift: My Site > Plugins (you'll need a site with plugin support)
✅ AbstractPostListViewController.swift: posts and pages list for a site.
✅ PostSettingsViewController.m: Edit a post > Tap "..." on the top right corner > Post Settings > Set Featured Image > "WordPress Media" album.
❌  PostGeolocationViewController.swift: The search field and the coloring here looks quite broken, but this wasn't introduced by this PR, so it's not actually a regression.
✅ ReaderFollowedSitesViewController.swift: Reader > Tap the gear in the top right corner  > "Followed Sites" tab
✅ ReaderSearchViewController.swift: Reader > Tap the magnifying lens on the top right corner
✅ ThemeBrowserViewController.swift: My Site > Themes
✅ TimeZoneSelectorViewController.swift: My Site > Site Settings > Timezone

3. What automated tests I added (or what prevented me from doing so)

None because this change is not easy to automatically test against.

## PR submission checklist:
